src/Uno.UI.RemoteControl.VS/EntryPoint.cs

```diff
@@ -534,21 +539,87 @@ private async Task CreateInfoBarAsync(NotificationRequestIdeMessage e, IVsShell
 		}
 	}
 
-	private async Task OnForceHotReloadRequestedAsync(object? sender, ForceHotReloadIdeMessage request)
+	private async Task OnForceHotReloadRequestedAsync(ForceHotReloadIdeMessage request)
 	{
 		try
 		{
+			// Programmatically trigger the "Apply Code Changes" command in Visual Studio.
+			// Which will trigger the hot reload.
 			_dte.ExecuteCommand("Debug.ApplyCodeChanges");
 
 			// Send a message back to indicate that the request has been received and acted upon.
 			if (_ideChannelClient is not null)
 			{
-				await _ideChannelClient.SendToDevServerAsync(new HotReloadRequestedIdeMessage(request.CorrelationId, Result.Success()), _ct.Token);
+				await _ideChannelClient.SendToDevServerAsync(new IdeResultMessage(request.CorrelationId, Result.Success()), _ct.Token);
+			}
+		}
+		catch (Exception e) when (_ideChannelClient is not null)
+		{
+			await _ideChannelClient.SendToDevServerAsync(new IdeResultMessage(request.CorrelationId, Result.Fail(e)), _ct.Token);
+
+			throw;
+		}
+	}
+
+	private async Task OnUpdateFileRequestedAsync(UpdateFileIdeMessage request)
+	{
+		try
+		{
+			if (request.FileContent is { Length: > 0 } fileContent)
+			{
+				var filePath = request.FileFullName;
+
+				// Update the file content in the IDE using the DTE API.
+				var document = _dte2.Documents
+					.OfType<Document>()
+					.FirstOrDefault(d => AbsolutePathComparer.ComparerIgnoreCase.Equals(d.FullName, filePath));
+
+				var textDocument = document?.Object("TextDocument") as TextDocument;
+
+				if (textDocument is null) // The document is not open in the IDE, so we need to open it.
```

Comment:
Why do we need to open it if it's currently closed? This seems unnecessary?

Reply:
To save it through the IDE. To simplify, all saves at done through the IDE instead of just those who at opened.